src/main/java/com/exasol/adapter/jdbc/AbstractRemoteMetadataReader.java

```diff
@@ -91,22 +91,39 @@ public IdentifierConverter getIdentifierConverter() {
 
     @Override
     public SchemaMetadata readRemoteSchemaMetadata() {
+        return readRemoteSchemaMetadataForSelectedTables(Collections.emptyList());
+    }
+
+    @Override
+    public SchemaMetadata readRemoteSchemaMetadata(final List<String> selectedTables) {
+        return readRemoteSchemaMetadataForSelectedTables(selectedTables);
+    }
+
+    private SchemaMetadata readRemoteSchemaMetadataForSelectedTables(final List<String> selectedTables) {
         try {
-            final String adapterNotes = SchemaAdapterNotesJsonConverter.getInstance()
-                    .convertToJson(getSchemaAdapterNotes());
-            final DatabaseMetaData remoteMetadata = this.connection.getMetaData();
-            final List<TableMetadata> tables = extractTableMetadata(remoteMetadata, Optional.empty());
-            return new SchemaMetadata(adapterNotes, tables);
+            return this.getSchemaMetadata(selectedTables);
         } catch (final SQLException exception) {
-            throw new RemoteMetadataReaderException(
-                    "Unable to read remote schema metadata. SQL error: " + exception.getMessage(), exception);
+            throw new RemoteMetadataReaderException("Unable to read remote schema metadata.", exception);
```

Comment:
Don't remove the cause workaround. Older versions of the UDF don't display the exception stack.